src/socket.ts

```diff
@@ -319,7 +328,13 @@ export class ConstellationSocket extends EventEmitter {
     }
 
     private rebroadcastEvent(name: string) {
-        this.socket.addEventListener(name, evt => this.emit(name, evt));
+        const socket = this.socket;
+        socket.addEventListener(name, evt => {
+            // Sanity check that the socket is current.
+            if (this.socket === socket) {
```

Comment:
Maybe it would be better make the client to not be able to reconnect until the previous socket has emitted its `close` event? Conceptually it's a bit more tidy.